### PR TITLE
feat(wallet/backend): add view card details endpoint

### LINKED ISSUE #1626: [Cards] Wallet Backend - View card details
### Context

Implement backend logic to view card details

### Todos

- [ ] controller
- [ ] service
- [ ] session management logic for viewing card details for a limited period of time
- [ ] tests

### PR DESCRIPTION
**Pull request by sanducb**
### Context

- fixes #1626 

### Changes

This PR adds endpoints for getting all cards for a specific customer and full card details for a specific card.

**Notes:**

- Keypair needed for `getCardDetails` endpoint will be generated and stored in the frontend, therefore `publicKeyBase64` will be sent on the request to wallet backend.
- Card details session will be handled in a different PR and ticket

### CODE REVIEW

File: packages/wallet/backend/src/card/controller.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  ) {}
+
+  public getMaskedCardDetails = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    try {
+      const userId = req.session.user.id
+      const { cardId } = req.params
```
Comment (beniaminmunteanu) [inline packages/wallet/backend/src/card/controller.ts:26]: request schema

File: packages/wallet/backend/src/card/controller.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  public getCardDetails = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    try {
+      const userId = req.session.user.id
+      const { cardId, publicKeyBase64 } = req.params
```
Comment (beniaminmunteanu) [inline packages/wallet/backend/src/card/controller.ts:52]: ditto

**Review by beniaminmunteanu — DISMISSED** [review]
LGTM. 
Would just add validation schemas and unit tests to verify throwing of 400 bad requests afterwards

**Review by beniaminmunteanu — APPROVED** [review]
🦁

File: packages/wallet/backend/src/app.ts [hunk 3]
```diff
@@ -301,6 +306,10 @@ export class App {
       gateHubController.addUserToGateway
     )
 
+    // Cards
+    router.get('/:customerId/cards', isAuth, cardController.getCardsByCustomer)
```
Comment (dragosp1011) [inline packages/wallet/backend/src/app.ts:310]: customers/:customerId/cards

File: packages/wallet/backend/src/gatehub/client.ts [hunk 4]
```diff
@@ -301,19 +310,74 @@ export class GateHubClient {
     return flatRates
   }
 
+  // This should be called before creating customers to get the product codes for the card and account
+  async fetchCardApplicationProducts(): Promise<ICardProductResponse[]> {
+    const path = `/v1/card-applications/${this.env.GATEHUB_CARD_APP_ID}/card-products`
```
Comment (dragosp1011) [inline packages/wallet/backend/src/gatehub/client.ts:315]: missed api url

File: packages/wallet/backend/src/gatehub/client.ts [hunk 5]
```diff
@@ -301,19 +310,74 @@ export class GateHubClient {
...
+    const url = `${this.apiUrl}/v1/customers`
+    return this.request<ICreateCustomerResponse>(
+      'POST',
+      url,
+      JSON.stringify(request)
+    )
+  }
+
+  async getCardsByCustomer(customerId: string): Promise<ICardResponse[]> {
+    const url = `/v1/customers/${customerId}/cards`
```
Comment (dragosp1011) [inline packages/wallet/backend/src/gatehub/client.ts:332]: missed api url

File: packages/wallet/backend/src/gatehub/client.ts [hunk 6]
```diff
@@ -348,14 +412,19 @@ export class GateHubClient {
...
+    managedUserUuid?: string,
+    token?: string
   ) {
     return {
       'Content-Type': 'application/json',
       'x-gatehub-app-id': this.env.GATEHUB_ACCESS_KEY,
       'x-gatehub-timestamp': timestamp,
       'x-gatehub-signature': this.getSignature(timestamp, method, url, body),
-      ...(managedUserUuid && { 'x-gatehub-managed-user-uuid': managedUserUuid })
+      'x-gatehub-card-app-id': this.env.GATEHUB_CARD_APP_ID,
```
Comment (dragosp1011) [inline packages/wallet/backend/src/gatehub/client.ts:423]: should this be included in each request?

Comment (sanducb) [reply packages/wallet/backend/src/gatehub/client.ts:423]: You're right, we only need this for customer and cards related requests.

Comment (dragosp1011) [reply packages/wallet/backend/src/gatehub/client.ts:436]: missed to remove `x-gatehub-card-app-id` here

**Review by beniaminmunteanu — APPROVED** [review]

**Pull request merged**